File: model/src/main/java/com/hexadeventure/model/map/CellType.java
package com.hexadeventure.model.map;

public enum CellType {
    GROUND,
    PATH,
    WALL;
    
    public static boolean isWalkable(CellType cellType) {
        return cellType != WALL;
    }
    
    public static int getCost(CellType cellType) {
        return switch (cellType) {
            case PATH -> 1;
            case GROUND -> 2;
            case WALL -> 200;
        };
    }
}

File: hexaDEVenture-unity/Assets/Scripts/Network/In/Game/DTOEnum/CellType.cs
namespace Network.In.Game.DTOEnum {
    public enum CellType {
        Ground,
        Ground2,
        Path,
        Wall

        /*public static boolean isWalkable(CellType cellType) {
        return cellType != WALL;
    }

    public static int getCost(CellType cellType, boolean onlyWalkable) {
        return switch (cellType) {
            case PATH -> 1;
            case GROUND -> 5;
            case GROUND2 -> 10;
            case WALL -> onlyWalkable ? -1 : 200;
        };
    }*/
    }
}